cpp/map_module/algorithm/map.cpp

```diff
@@ -1 +1,51 @@
 #include "map.hpp"
+
+void Map::removeRecursion(mgp::Map &result, bool recursive, std::string_view key) {
+  for (auto element : result) {
+    if (element.key == key) {
+      result.Erase(element.key);
+      continue;
+    }
+    if (element.value.IsMap() && recursive) {
+      mgp::Map non_const_value_map = mgp::Map(std::move(element.value.ValueMap()));
```

Comment:
I think @mpintaric55334 updated this part that we will use non const, so I will wait with this PR for change to land